src/vs/workbench/services/suggest/browser/simpleSuggestWidget.ts

```diff
@@ -439,8 +439,13 @@ export class SimpleSuggestWidget<TModel extends SimpleCompletionModel<TItem>, TI
 	}
 
 	private _setState(state: State): void {
-
 		if (this._state === state) {
+			if (this._state === State.Empty) {
+				// Don't keep the widget around if there are no suggestions
+				// as typing more is not going to yield new ones
+				this.hide();
+				this._state = State.Empty;
```

Comment:
Some recursion that complicated things here:

`_setState(State.Empty)` calls into hide which called `_setState(State.Hidden)`

`showSuggestions` has `isAuto` which I don't think we use but all it does is determines whether to use `Hidden` or `Empty`? Do we need to fill in `isAuto` to fix this instead?